tests/Imagine/Test/Image/AbstractImageTest.php

```diff
@@ -284,14 +284,22 @@ public function testThumbnailShouldReturnACopy()
         $this->assertNotSame($image, $thumbnail);
     }
 
-    public function testThumbnailWithInvalidModeShouldThrowAnException()
+    public function testThumbnailWithInvalidSettingShouldThrowAnException()
     {
         $factory = $this->getImagine();
         $image = $factory->open('tests/Imagine/Fixtures/google.png');
         $this->{method_exists($this, $_ = 'expectException') ? $_ : 'setExpectedException'}('Imagine\Exception\InvalidArgumentException', 'Invalid mode specified');
```

Comment:
You missed a "s/mode/setting" here :)

Reply:
@vlakoff Thanks. BTW any reason to rename the `$mode` argument to `$setting`? I think that this argument still defines the resize *mode*...

Reply:
That's because it is broader.

Previously, we could only choose between one mode or the other.

Now we can specify one or several settings, combining them. One mode or the other, and an option to allow upscaling (which isn't a mode).

Reply:
BTW it's main purpose still hasn't changed: it still defines how the image should be resized, isn't it?
If so, IMHO a parameter that describes how to do something is still a mode...